### DISCUSSION TITLE
Route Middleware

### DISCUSSION THREAD

**Discussion by riteshapatel** [sinatra/sinatra#1995] [Q&A]
Does `Sinatra` support `middleware` at `route level`? For example, in NodeJS I can do as below.

```
router.post("/like", validate(generic.FieldValueSchema), catchIt(controller.searchLike))

# validate is simply a generic payload / schema validator
```

In the event of a wrong payload I want to fail fast and not even reach the route handler. 

Or what is the best way to do `route level schema validation` in Sinatra? 

Any thoughts?

**Comment by dentarg** [comment]
Not sure I really get what you mean. I think you need to dig into the source to find your answer if the documentation isn't clear enough for you. Reading docs and code for https://github.com/sinatra/mustermann may also help you.
